### PR TITLE
invoice: make invoice_payment hook a multi-user hook.

### PR DESCRIPTION
**Pull request by rustyrussell**
We register on it for offers, and without this nobody else can.

Changelog-Changed: plugins: more than one plugin can now register invoice_payment hook.

### CODE REVIEW

File: lightningd/invoice.c [hunk 1]
```diff
@@ -393,6 +397,10 @@ void invoice_try_pay(struct lightningd *ld,
 	payload->set = set;
 	tal_add_destructor2(set, invoice_payload_remove_set, payload);
 
+	/* We notify here to benefit from the payload and because the hook callback is
```
Comment (ZmnSCPxj) [inline lightningd/invoice.c:400]: Comment no longer quite matches?  This is not the hook callback anymore, right?

Comment (rustyrussell) [reply lightningd/invoice.c:400]: Good point!

**Review by ZmnSCPxj — APPROVED** [review]
ACK 2dc52dbe47ceb16f73ddcb335132d3cebbf551c7

**Comment by rustyrussell** [comment]
Trivial fold-fixup-and-rebase-on-master.

**Pull request merged**